### ISSUE TITLE
Out of memory error....

### ISSUE THREAD

**Issue by shinahran** [google/ExoPlayer#1549]
I played local video to use ExtractorRendererBuilder.java. 
but It shown 'out of memory' when large video. (mp4, 300mb, 10min)
(I use Low spec set top box.) What is happen?

05-24 11:04:03.944 134-836/? D/OmxComponentManagerImpl: getEntryByName_2_num=0, componentName:OMX.amlogic.avc.decoder.awesome
05-24 11:04:03.944 134-836/? D/OmxComponentManagerImpl: amstream_vbuf is free
05-24 11:04:03.944 134-836/? D/OmxComponentManagerImpl: ret =0,*pEntry->mRoleNames=video_decoder.avc.awesome
05-24 11:04:03.944 134-836/? D/OmxComponentManagerImpl: ionvide is free
05-24 11:04:03.944 134-836/? D/OmxComponentManagerImpl: ready for creat omx decoder component
05-24 11:04:03.984 134-836/? V/AmlogicVideoDecoder: media.omx.latency is not set used def=0.600000
05-24 11:04:03.984 134-836/? E/OmxLogConf: Can not read property media.omx.log_levels, using 255
05-24 11:04:03.984 134-836/? D/OmxComponentManagerImpl: IncreaseEntryNumByName pEntry->mNum=0, pEntry->mMaxNum:4,componentName:OMX.amlogic.avc.decoder.awesome
05-24 11:04:03.984 134-836/? I/omx_core: OMX_getHandle_num=1
05-24 11:04:03.984 134-134/? V/OmxVideoDecoder: getParameter 257 nParamIndex=0x600000f
05-24 11:04:03.984 134-836/? V/OmxVideoDecoder: getParameter 257 nParamIndex=0x6000001
05-24 11:04:03.984 134-633/? V/OmxVideoDecoder: getParameter 257 nParamIndex=0x6000001
05-24 11:04:03.984 134-134/? D/OmxComponent: getExtensionIndex 1020 name="OMX.google.android.index.storeMetaDataInBuffers"
05-24 11:04:03.984 134-134/? D/OmxVideoDecoder: setParameter 457 OMX_IndexAndroidStoreMetadataInBuffers
05-24 11:04:03.984 134-134/? E/OMXNodeInstance: OMX_SetParameter() failed for StoreMetaDataInBuffers: 0x80001001
05-24 11:04:03.984 134-836/? D/OmxComponent: getExtensionIndex 1020 name="OMX.google.android.index.prepareForAdaptivePlayback"
05-24 11:04:03.984 134-836/? W/OMXNodeInstance: OMX_GetExtensionIndex OMX.google.android.index.prepareForAdaptivePlayback failed
05-24 11:04:03.984 134-633/? D/OmxComponentManagerImpl: DecreaseEntryNumByName pEntry->mNum=1, pEntry->mMaxNum:4,componentName:OMX.amlogic.avc.decoder.awesome
05-24 11:04:03.984 134-633/? I/omx_core: OMX_freeHandle_num_=0
05-24 11:04:03.984 2379-2440/com.app.media I/OMXClient: Using client-side OMX mux.
05-24 11:04:04.014 2379-2440/com.app.media I/omx_core: OMX_Init 66 gInitCount=1
05-24 11:04:04.014 2379-2440/com.app.media V/AML_all_components: --------------------------------
                                                                                  author:aml.sh multi-media team
                                                                                  branch name:   
                                                                                  git version:    
                                                                                  last changed:  
                                                                                  build-time:     Wed Aug 13 20:07:59 CST 2014
                                                                                  build-name:     brian.zhu
                                                                                  uncommitted-file-num:0
                                                                                  cd /mnt/fileroot/brian.zhu/m8baby
                                                                                  hostname droid05
                                                                                  --------------------------------
05-24 11:04:04.014 2379-2440/com.app.media V/OmxComponentManagerImpl: loadLibrary:225 Loading OMX library libOmxVideo.so.
05-24 11:04:04.014 2379-2440/com.app.media I/omx_core: Core library has 20 components
05-24 11:04:04.014 2379-2440/com.app.media E/OMXMaster: A component of name 'OMX.google.alac.decoder' already exists, ignoring this one.
05-24 11:04:04.024 2379-2440/com.app.media W/ALooperRoster: ALooper::handler_id ALooperRoster::registerHandler 1
05-24 11:04:04.024 2379-2440/com.app.media I/AmSoftDec: get mimetype : video/avc
05-24 11:04:04.024 2379-2440/com.app.media I/AmFFmpegCodec: CreateCodec for codecMime : video/avc
05-24 11:04:04.094 2379-2440/com.app.media E/OMXNodeInstance: OMX_GetExtensionIndex OMX.google.android.index.storeMetaDataInBuffers failed
05-24 11:04:04.094 2379-2440/com.app.media W/OMXNodeInstance: OMX_GetExtensionIndex OMX.google.android.index.prepareForAdaptivePlayback failed
05-24 11:04:04.094 2379-2440/com.app.media I/ALooperRoster: ALooperRoster unregisterHandler handlerID=1
05-24 11:04:04.094 2379-2440/com.app.media I/omx_core: OMX_Deinit 96 gInitCount=0
05-24 11:04:04.104 2379-2440/com.app.media I/11: >>onDiscontinuity
05-24 11:04:04.104 2379-2379/com.app.media I/1: <>>> onPlayerStateChanged playWhenReady=true > playbackState=3
05-24 11:04:04.104 2379-2379/com.app.media I/1: <>>> onPlayerStateChanged STATE_BUFFERING 0  >  -1
05-24 11:04:04.104 2379-2379/com.app.media I/1: >>> startBufferTimer > bufferdPercent=0
05-24 11:04:04.104 2379-2440/com.app.media W/ALooperRoster: ALooper::handler_id ALooperRoster::registerHandler 2
05-24 11:04:04.104 2379-2440/com.app.media W/ALooperRoster: ALooper::handler_id ALooperRoster::registerHandler 3
05-24 11:04:04.114 2379-2484/com.app.media I/OMXClient: Using client-side OMX mux.
05-24 11:04:04.114 134-633/? D/OmxComponentManagerImpl: getEntryByName_2_num=0, componentName:OMX.amlogic.avc.decoder.awesome
05-24 11:04:04.114 134-633/? D/OmxComponentManagerImpl: amstream_vbuf is free
05-24 11:04:04.114 134-633/? D/OmxComponentManagerImpl: ret =0,*pEntry->mRoleNames=video_decoder.avc.awesome
05-24 11:04:04.114 134-633/? D/OmxComponentManagerImpl: ionvide is free
05-24 11:04:04.114 134-633/? D/OmxComponentManagerImpl: ready for creat omx decoder component
05-24 11:04:04.114 134-633/? V/AmlogicVideoDecoder: media.omx.latency is not set used def=0.600000
05-24 11:04:04.114 134-633/? E/OmxLogConf: Can not read property media.omx.log_levels, using 255
05-24 11:04:04.114 134-633/? D/OmxComponentManagerImpl: IncreaseEntryNumByName pEntry->mNum=0, pEntry->mMaxNum:4,componentName:OMX.amlogic.avc.decoder.awesome
05-24 11:04:04.114 134-633/? I/omx_core: OMX_getHandle_num=1
05-24 11:04:04.114 134-836/? D/OmxComponent: getExtensionIndex 1020 name="OMX.google.android.index.storeMetaDataInBuffers"
05-24 11:04:04.114 134-836/? D/OmxVideoDecoder: setParameter 457 OMX_IndexAndroidStoreMetadataInBuffers
05-24 11:04:04.114 134-836/? E/OMXNodeInstance: OMX_SetParameter() failed for StoreMetaDataInBuffers: 0x80001001
05-24 11:04:04.114 2379-2484/com.app.media E/ACodec: [OMX.amlogic.avc.decoder.awesome] storeMetaDataInBuffers failed w/ err -2147483648
05-24 11:04:04.124 134-633/? V/OmxVideoDecoder: getParameter 257 nParamIndex=0x6000001
05-24 11:04:04.124 134-134/? D/OmxVideoDecoder: setParameter nPortIndex 0 CompFmt 0x7 ColorFmt 0
05-24 11:04:04.124 134-836/? V/OmxVideoDecoder: getParameter 257 nParamIndex=0x6000001
05-24 11:04:04.124 134-633/? D/OmxVideoDecoder: setParameter nPortIndex 1 CompFmt 0 ColorFmt 0x15
05-24 11:04:04.124 134-134/? V/OmxVideoDecoder: getParameter 257 nParamIndex=0x2000001
05-24 11:04:04.124 134-836/? D/OmxVideoDecoder: setParameter nPortIndex 0 nFrameWidth 1920 nFrameHeight 1080 nSliceHeight 16
05-24 11:04:04.124 134-836/? E/OmxVideoDecoder:  reset input buffer:1036800
05-24 11:04:04.124 134-633/? V/OmxVideoDecoder: getParameter 257 nParamIndex=0x2000001
05-24 11:04:04.124 134-633/? V/OmxVideoDecoder: mUseNativeBuffers 0
05-24 11:04:04.124 134-633/? V/OmxVideoDecoder: port.mDef.nBufferSize 0
05-24 11:04:04.124 134-134/? D/OmxVideoDecoder: setParameter nPortIndex 1 nFrameWidth 1920 nFrameHeight 1080 nSliceHeight 16
05-24 11:04:04.124 134-134/? V/OmxVideoDecoder: mUseNativeBuffers 0
05-24 11:04:04.124 134-134/? V/OmxVideoDecoder: port.mDef.nBufferSize 3110400
05-24 11:04:04.124 134-836/? V/OmxVideoDecoder: getParameter 257 nParamIndex=0x2000001
05-24 11:04:04.124 2379-2484/com.app.media I/ACodec: onConfigureComponent native-window finished OMX.amlogic.avc.decoder.awesome 
05-24 11:04:04.124 134-633/? D/OmxComponent: getExtensionIndex 1020 name="OMX.google.android.index.enableAndroidNativeBuffers"
05-24 11:04:04.124 134-633/? V/OmxVideoDecoder: setParameter 451 EnableNativeBuffers 1
05-24 11:04:04.124 134-633/? V/OmxVideoDecoder: mUseNativeBuffers 1
05-24 11:04:04.124 134-633/? D/AmlogicVideoDecoderAwesome: get native buffer info NV:w:1920,h:1080..
05-24 11:04:04.124 134-633/? V/OmxVideoDecoder: updatePort:portDef.format.video.nFrameWidth=1920,portDef.format.video.nFrameHeight=1080
05-24 11:04:04.124 134-633/? V/OmxVideoDecoder: port.mDef.nBufferSize 3110400
05-24 11:04:04.124 134-836/? V/OmxVideoDecoder: getParameter 257 nParamIndex=0x2000001
05-24 11:04:04.134 134-836/? V/OmxVideoDecoder: getParameter 257 nParamIndex=0x2000001
05-24 11:04:04.134 134-836/? V/OmxVideoDecoder: mUseNativeBuffers 1
05-24 11:04:04.134 134-836/? D/AmlogicVideoDecoderAwesome: get native buffer info NV:w:1920,h:1080..
05-24 11:04:04.134 134-836/? V/OmxVideoDecoder: updatePort:portDef.format.video.nFrameWidth=1920,portDef.format.video.nFrameHeight=1080
05-24 11:04:04.134 134-836/? V/OmxVideoDecoder: port.mDef.nBufferSize 3110400
05-24 11:04:04.134 134-633/? D/OmxComponent: getExtensionIndex 1020 name="OMX.google.android.index.getAndroidNativeBufferUsage"
05-24 11:04:04.134 134-633/? V/OmxVideoDecoder: getParameter 257 nParamIndex=0x7f000001
05-24 11:04:04.134 134-134/? D/OmxVideoDecoder: setParameter nPortIndex 1 nFrameWidth 1920 nFrameHeight 1080 nSliceHeight 16
05-24 11:04:04.134 134-134/? V/OmxVideoDecoder: mUseNativeBuffers 1
05-24 11:04:04.134 134-134/? D/AmlogicVideoDecoderAwesome: get native buffer info NV:w:1920,h:1080..
05-24 11:04:04.134 134-134/? V/OmxVideoDecoder: updatePort:portDef.format.video.nFrameWidth=1920,portDef.format.video.nFrameHeight=1080
05-24 11:04:04.134 134-134/? V/OmxVideoDecoder: port.mDef.nBufferSize 3110400
05-24 11:04:04.134 134-836/? D/OmxComponent: getExtensionIndex 1020 name="OMX.google.android.index.useAndroidNativeBuffer2"
05-24 11:04:04.134 134-836/? D/OmxComponent: getExtensionIndex 1020 name="OMX.google.android.index.useAndroidNativeBuffer"
05-24 11:04:04.134 134-633/? D/OmxComponent: getExtensionIndex 1020 name="OMX.google.android.index.useAndroidNativeBuffer2"
05-24 11:04:04.134 134-633/? D/OmxComponent: getExtensionIndex 1020 name="OMX.google.android.index.useAndroidNativeBuffer"
05-24 11:04:04.134 134-134/? D/OmxComponent: getExtensionIndex 1020 name="OMX.google.android.index.useAndroidNativeBuffer2"
05-24 11:04:04.134 134-134/? D/OmxComponent: getExtensionIndex 1020 name="OMX.google.android.index.useAndroidNativeBuffer"
05-24 11:04:04.134 134-836/? D/OmxComponent: getExtensionIndex 1020 name="OMX.google.android.index.useAndroidNativeBuffer2"
05-24 11:04:04.134 134-836/? D/OmxComponent: getExtensionIndex 1020 name="OMX.google.android.index.useAndroidNativeBuffer"
05-24 11:04:04.134 134-633/? D/OmxComponent: getExtensionIndex 1020 name="OMX.google.android.index.useAndroidNativeBuffer2"
05-24 11:04:04.134 134-633/? D/OmxComponent: getExtensionIndex 1020 name="OMX.google.android.index.useAndroidNativeBuffer"
05-24 11:04:04.144 131-2012/? E/ion: ioctl c0144900 failed with code -1: Out of memory
05-24 11:04:04.144 131-2012/? E/[Gralloc-ERROR]: int gralloc_alloc_buffer(alloc_device_t*, size_t, int, const native_handle_t**):115 Failed to ion_alloc from ion_client:20
05-24 11:04:04.144 131-2012/? W/GraphicBufferAllocator: alloc(1920, 1080, 17, 20002100, ...) failed -1 (Operation not permitted)
                                                        
                                                        [ 05-24 11:04:04.144   131: 2012 E/         ]
                                                        GraphicBufferAlloc::createGraphicBuffer(w=1920, h=1080) failed (Operation not permitted), handle=0x0
05-24 11:04:04.144 2379-2484/com.app.media E/BufferQueue: [unnamed-2379-1] dequeueBuffer: SurfaceComposer::createGraphicBuffer failed
05-24 11:04:04.144 2379-2484/com.app.media E/ACodec: dequeueBuffer failed: Operation not permitted (1)
05-24 11:04:04.144 2379-2484/com.app.media E/ACodec: Failed to allocate buffers after transitioning to IDLE state (error 0xffffffff)
05-24 11:04:04.144 2379-2483/com.app.media E/MediaCodec: Codec reported an error. (omx error 0x80001001, internalError -1)
05-24 11:04:04.144 2379-2440/com.app.media E/dalvikvm: Could not find class 'android.media.MediaCodec$CodecException', referenced from method com.google.android.exoplayer.MediaCodecTrackRenderer$DecoderInitializationException.getDiagnosticInfoV21
05-24 11:04:04.144 2379-2440/com.app.media W/dalvikvm: VFY: unable to resolve instanceof 128 (Landroid/media/MediaCodec$CodecException;) in Lcom/google/android/exoplayer/MediaCodecTrackRenderer$DecoderInitializationException;
05-24 11:04:04.144 2379-2440/com.app.media D/dalvikvm: VFY: replacing opcode 0x20 at 0x0000
05-24 11:04:04.144 2379-2440/com.app.media E/ExoPlayerImplInternal: Internal track renderer error.
                                                                                     com.google.android.exoplayer.ExoPlaybackException: com.google.android.exoplayer.MediaCodecTrackRenderer$DecoderInitializationException: Decoder init failed: OMX.amlogic.avc.decoder.awesome, MediaFormat(1, video/avc, -1, 876642, 1920, 1080, 0, 1.0, -1, -1, null, 634533333, false, -1, -1, -1, -1, -1)
                                                                                         at com.google.android.exoplayer.MediaCodecTrackRenderer.notifyAndThrowDecoderInitError(MediaCodecTrackRenderer.java:416)
                                                                                         at com.google.android.exoplayer.MediaCodecTrackRenderer.maybeInitCodec(MediaCodecTrackRenderer.java:402)
                                                                                         at com.google.android.exoplayer.MediaCodecTrackRenderer.onInputFormatChanged(MediaCodecTrackRenderer.java:776)
                                                                                         at com.google.android.exoplayer.MediaCodecVideoTrackRenderer.onInputFormatChanged(MediaCodecVideoTrackRenderer.java:333)
                                                                                         at com.google.android.exoplayer.MediaCodecTrackRenderer.readFormat(MediaCodecTrackRenderer.java:530)
                                                                                         at com.google.android.exoplayer.MediaCodecTrackRenderer.doSomeWork(MediaCodecTrackRenderer.java:512)
                                                                                         at com.google.android.exoplayer.MediaCodecVideoTrackRenderer.doSomeWork(MediaCodecVideoTrackRenderer.java:621)
                                                                                         at com.google.android.exoplayer.SampleSourceTrackRenderer.doSomeWork(SampleSourceTrackRenderer.java:129)
                                                                                         at com.google.android.exoplayer.ExoPlayerImplInternal.doSomeWork(ExoPlayerImplInternal.java:434)
                                                                                         at com.google.android.exoplayer.ExoPlayerImplInternal.handleMessage(ExoPlayerImplInternal.java:213)
                                                                                         at android.os.Handler.dispatchMessage(Handler.java:98)
                                                                                         at android.os.Looper.loop(Looper.java:136)
                                                                                         at android.os.HandlerThread.run(HandlerThread.java:61)
                                                                                         at com.google.android.exoplayer.util.PriorityHandlerThread.run(PriorityHandlerThread.java:40)
                                                                                      Caused by: com.google.android.exoplayer.MediaCodecTrackRenderer$DecoderInitializationException: Decoder init failed: OMX.amlogic.avc.decoder.awesome, MediaFormat(1, video/avc, -1, 876642, 1920, 1080, 0, 1.0, -1, -1, null, 634533333, false, -1, -1, -1, -1, -1)
                                                                                         at com.google.android.exoplayer.MediaCodecTrackRenderer.maybeInitCodec(MediaCodecTrackRenderer.java:402) 
                                                                                         at com.google.android.exoplayer.MediaCodecTrackRenderer.onInputFormatChanged(MediaCodecTrackRenderer.java:776) 
                                                                                         at com.google.android.exoplayer.MediaCodecVideoTrackRenderer.onInputFormatChanged(MediaCodecVideoTrackRenderer.java:333) 
                                                                                         at com.google.android.exoplayer.MediaCodecTrackRenderer.readFormat(MediaCodecTrackRenderer.java:530) 
                                                                                         at com.google.android.exoplayer.MediaCodecTrackRenderer.doSomeWork(MediaCodecTrackRenderer.java:512) 
                                                                                         at com.google.android.exoplayer.MediaCodecVideoTrackRenderer.doSomeWork(MediaCodecVideoTrackRenderer.java:621) 
                                                                                         at com.google.android.exoplayer.SampleSourceTrackRenderer.doSomeWork(SampleSourceTrackRenderer.java:129) 
                                                                                         at com.google.android.exoplayer.ExoPlayerImplInternal.doSomeWork(ExoPlayerImplInternal.java:434) 
                                                                                         at com.google.android.exoplayer.ExoPlayerImplInternal.handleMessage(ExoPlayerImplInternal.java:213) 
                                                                                         at android.os.Handler.dispatchMessage(Handler.java:98) 
                                                                                         at android.os.Looper.loop(Looper.java:136) 
                                                                                         at android.os.HandlerThread.run(HandlerThread.java:61) 
                                                                                         at com.google.android.exoplayer.util.PriorityHandlerThread.run(PriorityHandlerThread.java:40) 
                                                                                      Caused by: java.lang.IllegalStateException: start failed
                                                                                         at android.media.MediaCodec.start(Native Method)
                                                                                         at com.google.android.exoplayer.MediaCodecTrackRenderer.maybeInitCodec(MediaCodecTrackRenderer.java:393)
                                                                                         at com.google.android.exoplayer.MediaCodecTrackRenderer.onInputFormatChanged(MediaCodecTrackRenderer.java:776) 
                                                                                         at com.google.android.exoplayer.MediaCodecVideoTrackRenderer.onInputFormatChanged(MediaCodecVideoTrackRenderer.java:333) 
                                                                                         at com.google.android.exoplayer.MediaCodecTrackRenderer.readFormat(MediaCodecTrackRenderer.java:530) 
                                                                                         at com.google.android.exoplayer.MediaCodecTrackRenderer.doSomeWork(MediaCodecTrackRenderer.java:512) 
                                                                                         at com.google.android.exoplayer.MediaCodecVideoTrackRenderer.doSomeWork(MediaCodecVideoTrackRenderer.java:621) 
                                                                                         at com.google.android.exoplayer.SampleSourceTrackRenderer.doSomeWork(SampleSourceTrackRenderer.java:129) 
                                                                                         at com.google.android.exoplayer.ExoPlayerImplInternal.doSomeWork(ExoPlayerImplInternal.java:434) 
                                                                                         at com.google.android.exoplayer.ExoPlayerImplInternal.handleMessage(ExoPlayerImplInternal.java:213) 
                                                                                         at android.os.Handler.dispatchMessage(Handler.java:98) 
                                                                                         at android.os.Looper.loop(Looper.java:136) 
                                                                                         at android.os.HandlerThread.run(HandlerThread.java:61) 
                                                                                         at com.google.android.exoplayer.util.PriorityHandlerThread.run(PriorityHandlerThread.java:40) 
05-24 11:04:04.144 2379-2379/com.app.media I/1: >>>onPlayerError
05-24 11:04:04.154 134-134/? D/OmxComponentManagerImpl: DecreaseEntryNumByName pEntry->mNum=1, pEntry->mMaxNum:4,componentName:OMX.amlogic.avc.decoder.awesome
05-24 11:04:04.154 134-134/? I/omx_core: OMX_freeHandle_num_=0
05-24 11:04:04.154 2379-2379/com.app.media I/1: <>>> onPlayerStateChanged playWhenReady=true > playbackState=1

**Comment by samkazmi** [comment]
```
E/LoadTask: Unexpected error loading stream
                                                                java.lang.OutOfMemoryError
                                                                    at java.lang.String.<init>(String.java:422)
                                                                    at java.io.BufferedReader.readLine(BufferedReader.java:363)
                                                                    at com.google.android.exoplayer.hls.HlsPlaylistParser$LineIterator.hasNext(HlsPlaylistParser.java:335)
                                                                    at com.google.android.exoplayer.hls.HlsPlaylistParser.parseMediaPlaylist(HlsPlaylistParser.java:251)
                                                                    at com.google.android.exoplayer.hls.HlsPlaylistParser.parse(HlsPlaylistParser.java:132)
                                                                    at com.google.android.exoplayer.hls.HlsChunkSource$MediaPlaylistChunk.consume(HlsChunkSource.java:915)
                                                                    at com.google.android.exoplayer.chunk.DataChunk.load(DataChunk.java:97)
                                                                    at com.google.android.exoplayer.upstream.Loader$LoadTask.run(Loader.java:209)
                                                                    at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:390)
                                                                    at java.util.concurrent.FutureTask.run(FutureTask.java:234)
                                                                    at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1080)
                                                                    at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:573)
                                                                    at java.lang.Thread.run(Thread.java:841)
07-02 21:07:58.486 19590-19709/com.pantera.pbchurch W/dalvikvm: threadid=27: thread exiting with uncaught exception (group=0x41bf2700)
07-02 21:07:58.491 19590-19709/com.pantera.pbchurch E/AndroidRuntime: FATAL EXCEPTION: ExoPlayerImplInternal:Handler
                                                                      java.lang.OutOfMemoryError
                                                                          at java.lang.String.<init>(String.java:422)
                                                                          at java.io.BufferedReader.readLine(BufferedReader.java:363)
                                                                          at com.google.android.exoplayer.hls.HlsPlaylistParser$LineIterator.hasNext(HlsPlaylistParser.java:335)
                                                                          at com.google.android.exoplayer.hls.HlsPlaylistParser.parseMediaPlaylist(HlsPlaylistParser.java:251)
                                                                          at com.google.android.exoplayer.hls.HlsPlaylistParser.parse(HlsPlaylistParser.java:132)
                                                                          at com.google.android.exoplayer.hls.HlsChunkSource$MediaPlaylistChunk.consume(HlsChunkSource.java:915)
                                                                          at com.google.android.exoplayer.chunk.DataChunk.load(DataChunk.java:97)
                                                                          at com.google.android.exoplayer.upstream.Loader$LoadTask.run(Loader.java:209)
                                                                          at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:390)
                                                                          at java.util.concurrent.FutureTask.run(FutureTask.java:234)
                                                                          at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1080)
                                                                          at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:573)
                                                                          at java.lang.Thread.run(Thread.java:841)
```

**Comment by steve111MV** [comment]
@samkazmi @shinahran  can you please share the solution of this ?
@willnorris @canatella @talklittle

**Comment by ojw28** [comment]
This error could be caused by more or less anything, including a memory leak in your own application code. The point at which an OutOfMemoryError is thrown is often not indicative of the root cause. You need to analyze your application's memory usage to determine what's happening. Some information on how to do this is provided [here](https://developer.android.com/studio/profile/investigate-ram.html).

**Comment by ojw28** [comment]
Actually, for this specific case it's likely you're hitting https://github.com/google/ExoPlayer/issues/2301, which was fixed in 2.2.0 and above. If you're already using 2.2.0 or greater, the advice above still holds.
